core/src/main/java/org/apache/druid/data/input/impl/CloudObjectLocation.java

```diff
@@ -71,7 +71,7 @@ public CloudObjectLocation(@JsonProperty("bucket") String bucket, @JsonProperty(
 
   public CloudObjectLocation(URI uri)
   {
-    this(uri.getHost(), uri.getPath());
+    this(uri.getAuthority(), uri.getPath());
```

Comment:
should we fall back to authority only if getHost() returns null?

Reply:
Yeah, I would be more comfortable with that too.

Reply:
That would work as well.